pkg/controller/common/helpers.go

```diff
@@ -325,26 +326,56 @@ func convertIgnition32to31(ign3config ign3types.Config) (ign3_1types.Config, err
 func ValidateIgnition(ignconfig interface{}) error {
 	switch cfg := ignconfig.(type) {
 	case ign2types.Config:
-		if reflect.DeepEqual(ign2types.Config{}, cfg) {
-			return nil
-		}
-		if report := validate2.ValidateWithoutSource(reflect.ValueOf(cfg)); report.IsFatal() {
-			return errors.Errorf("invalid ignition V2 config found: %v", report)
-		}
-		return nil
+		return validateIgn2(cfg)
 	case ign3types.Config:
-		if reflect.DeepEqual(ign3types.Config{}, cfg) {
-			return nil
-		}
-		if report := validate3.ValidateWithContext(cfg, nil); report.IsFatal() {
-			return errors.Errorf("invalid ignition V3 config found: %v", report)
-		}
-		return nil
+		return validateIgn3(cfg)
 	default:
 		return errors.Errorf("unrecognized ignition type")
 	}
 }
 
+// Validates Ignition V2 configs
+func validateIgn2(cfg ign2types.Config) error {
+	if reflect.DeepEqual(ign2types.Config{}, cfg) {
+		return nil
+	}
+
+	if report := validate2.ValidateWithoutSource(reflect.ValueOf(cfg)); report.IsFatal() {
+		return errors.Errorf("invalid ignition V2 config found: %v", report)
+	}
+
+	// Validates that file modes do not have special bits (sticky, setuid, setgid) set
+	// https://bugzilla.redhat.com/show_bug.cgi?id=2038240
+	for _, file := range cfg.Storage.Files {
```

Comment:
Could we pop these two funcs back into the switch, only part that could be its own new func is this looping over the cfg.storagefiles bit since it's shared code between the two (unless my tired eyes are deceiving me :smile_cat: )

Reply:
I wish I could share the loop / detection code between the v2 and v3 file objects. Unfortunately, they're each their own types. I  _could_ wrap the Ignition v2 / v3 File objects in an outer struct that satisfies a common interface, but I think that might be too complicated for this specific instance. As an aside, that could be an interesting refactoring idea for the Ignition v2 / v3 stuff, but that's far beyond the scope of this PR.

Getting back on-topic, what I'll do is pop that logic back into the `ValidateIgnition` function, then write separate functions (for the Ignition v2 / v3 cases) to handle the file mode checks.

Reply:
ah good point agree. at some point let's think about the v2/v3 refactor, bc it's kind of a drag to do everything 2x. 

sep file mode checks sound good. thanks!